File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        
        Dictionary<int, int> count = new();
        foreach(int num in nums)
        {
            count[num] = count.GetValueOrDefault(num) + 1;
        }
        List<int>[] bucket = new List<int>[nums.Length+1];

        foreach(var pair in count)
        {
            if(bucket[pair.Value] == null) bucket[pair.Value] = new List<int>();
            bucket[pair.Value].Add(pair.Key);
        }

        List<int> result = new();

        for(int i = bucket.Length-1; i >= 0; i--)
        {
            if(bucket[i] != null) result.AddRange(bucket[i]);
        }

        return result.Take(k).ToArray();
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-2.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        if(nums.size() <= 1) return 0;

        for(int i = 0, j = 0, cnt = 0; i < nums.size() && i <= j; )
        {
            int mx = j;
            while(i <= j)
            {
                mx = max(mx, i+nums[i]);
                i++;
            }
            if(mx > j) cnt++, j = mx;
            if(j >= nums.size()-1) return cnt;
        }
    }
};
